File: articles/behavioral-design-technology-systems/go/interface_model.go
package main

import (
	"fmt"
	"math"
)

func logistic(x float64) float64 {
	return 1.0 / (1.0 + math.Exp(-x))
}

func joinProbability(
	baselineValue float64,
	salienceSensitivity float64,
	defaultSensitivity float64,
	frictionSensitivity float64,
	rewardSensitivity float64,
	cognitiveOverload float64,
	salience float64,
	defaultOn float64,
	entryFriction float64,
	rewardIntensity float64,
) float64 {
	score := baselineValue +
		salienceSensitivity*salience +
		defaultSensitivity*defaultOn -
		frictionSensitivity*entryFriction +
		rewardSensitivity*rewardIntensity -
		cognitiveOverload*0.4

	return logistic(score)
}

func main() {
	p := joinProbability(0.45, 0.55, 0.50, 0.60, 0.58, 0.42, 0.55, 0.0, 0.08, 0.35)
	fmt.Printf("Synthetic join probability: %.3f\n", p)
}

File: articles/behavioral-design-technology-systems/fortran/interface_model.f90
program interface_model
  implicit none

  real :: baseline_value, salience_sensitivity, default_sensitivity
  real :: friction_sensitivity, reward_sensitivity, cognitive_overload
  real :: salience, default_on, entry_friction, reward_intensity
  real :: score, probability

  baseline_value = 0.45
  salience_sensitivity = 0.55
  default_sensitivity = 0.50
  friction_sensitivity = 0.60
  reward_sensitivity = 0.58
  cognitive_overload = 0.42
  salience = 0.55
  default_on = 0.0
  entry_friction = 0.08
  reward_intensity = 0.35

  score = baseline_value &
        + salience_sensitivity * salience &
        + default_sensitivity * default_on &
        - friction_sensitivity * entry_friction &
        + reward_sensitivity * reward_intensity &
        - cognitive_overload * 0.4

  probability = 1.0 / (1.0 + exp(-score))

  print *, "Synthetic join probability:", probability
end program interface_model
